runtime/service.go

```diff
@@ -1248,6 +1260,9 @@ func (s *service) shutdownLoop(
 		{
 			name: "send a request to the in-VM agent",
 			shutdown: func() error {
+				if s.agentClient == nil {
+					return nil
```

Comment:
Shouldn't be an error?

Reply:
I don't necessarily think so as this would just mean the `agentClient` wasn't even set. But I am indifferent on returning an error or not.

Reply:
Let's return an error then. We log all errors in the loop below. So this would be helpful for operators.

Reply:
Turns out all these nil checks were due to a previous change I had and is no longer needed. So I have removed them